pkg/kudoctl/cmd/params/parser.go

```diff
@@ -51,35 +51,42 @@ func getParamsFromFiles(fs afero.Fs, filePaths []string, errs []string) (map[str
 			errs = append(errs, fmt.Sprintf("error reading from parameter file %s: %v", filePath, err))
 			continue
 		}
-		data := make(map[string]interface{})
-		err = yaml.Unmarshal(rawData, &data)
+
+		errs = GetParametersFromFile(filePath, rawData, errs, parameters)
+
+	}
+	return parameters, errs
+}
+
+func GetParametersFromFile(filePath string, bytes []byte, errs []string, parameters map[string]string) []string {
+	data := make(map[string]interface{})
+	err := yaml.Unmarshal(bytes, &data)
+	if err != nil {
+		errs = append(errs, fmt.Sprintf("error unmarshalling content of parameter file %s: %v", filePath, err))
+		return errs
```

Comment:
not mentioned above... but it is also strange to pass in a ref to `errs []string` and also return that `[]string`

Reply:
prefer the following:

```
func GetParametersFromFile(filePath string, bytes []byte, parameters map[string]string) error {
	data := make(map[string]interface{})
	err := yaml.Unmarshal(bytes, &data)
	if err != nil {
		return fmt.Errorf("error unmarshalling content of parameter file %s: %w", filePath, err)
	}
	clog.V(2).Printf("Unmarshalling %q...", filePath)
	castErrors := []string{}
	for key, value := range data {
		clog.V(3).Printf("Value of parameter %q is a %T: %v", key, value, value)
		var valueType v1beta1.ParameterType
		switch value.(type) {
		case map[string]interface{}:
			valueType = v1beta1.MapValueType
		case []interface{}:
			valueType = v1beta1.ArrayValueType
		case string:
			valueType = v1beta1.StringValueType
		default:
			valueType = v1beta1.StringValueType
		}
		wrapped, err := convert.WrapParamValue(value, valueType)
		if err != nil {
			castErrors = append(castErrors, fmt.Sprintf("%s", key))
			//errs = append(errs, fmt.Sprintf("error converting value of parameter %s from file %s %q to a string: %v", key, filePath, value, err))
			continue
		}
		parameters[key] = *wrapped
	}
	if len(castErrors) > 0 {
		return fmt.Errorf("error converting values to strings from file %s for the following keys [%s]", strings.Join(castErrors, ", "))
	}
	return nil
}
```

Reply:
with details of err in a `clog.V(x)` log

Reply:
These are all valid points, but I merely extracted this part from the `getParamsFromFiles` method above and kept the overall structure. This is the way this parser is built starting with the entry method `GetParameterMap` above: all the helper methods are treating errors as strings gathering them together.

Imho the biggest drawback of the above approach is that the individual errors collected in this line:
```
errs = append(errs, fmt.Sprintf("error converting value of parameter %s from file %s %q to a string: %v", key, filePath, value, err))
```
are lost and replaced by a generic `error converting values to strings` without extra details of what exactly is the problem.

Reply:
Ok, I reworked it and the end result is pretty close to the variant above: verbosity is reduced and the method returns an `error`. However, I keep the original error with the corresponding key.